pyvo/dal/sia2.py

```diff
@@ -21,12 +25,12 @@
 from ..dam import ObsCoreMetadata, CALIBRATION_LEVELS
 
 
-__all__ = ["search", "SIAService", "SIAQuery", "SIAResults", "ObsCoreRecord"]
+__all__ = ["search", "SIA2Service", "SIA2Query", "SIA2Results", "ObsCoreRecord"]
```

Comment:
Those doing `import pyvo.dal.sia2.*` will be punished but this is a know anti-pattern so I'm OK with it.

Reply:
Yep, a but aggressive lesson on not to do * imports, however this is at quite a usual way of doing deprecations.

After merging this I'll look for use cases out in the wild and notify them as necessary